### PR TITLE
Moved grouping algorithm to front end

### PR DESCRIPTION
**Pull request by jtkpiotr**
@adamzima @s1mplex @pbzymek @lkudyba @quirk0o @mhib

### CODE REVIEW

File: client/app/components/calendar/Calendar.jsx [hunk 1]
```diff
@@ -12,7 +13,7 @@ const { string, number, array, arrayOf, oneOfType, instanceOf, func } = PropType
 
 export default class Calendar extends React.Component {
   static propTypes = {
-    events:                   array,
+    events:                 arrayOf(EventSchema.only('start_timestamp', 'end_timestamp', 'conference_room')).isRequired,
```
Comment (adamzima) [inline client/app/components/calendar/Calendar.jsx]: wrong indentation

File: client/app/components/calendar/CalendarRow.jsx [hunk 2]
```diff
@@ -9,7 +9,7 @@ const { string, bool, number, arrayOf, oneOfType, instanceOf, func } = React.Pro
 
 export default class CalendarRow extends React.Component {
   static propTypes = {
-    events:                   arrayOf(arrayOf(EventSchema.only('start'))).isRequired,
+    events:    arrayOf(EventSchema.only('start_timestamp')).isRequired,
```
Comment (adamzima) [inline client/app/components/calendar/CalendarRow.jsx]: wrong indentation

File: client/app/components/calendar/event/Event.jsx [hunk 3]
```diff
@@ -9,11 +9,12 @@ import './event.scss';
 
 export default class Event extends React.Component {
   static propTypes = {
-    event:                    EventSchema.isRequired,
-    containerHeight:          PropTypes.number.isRequired,
+    event: EventSchema.isRequired,
```
Comment (adamzima) [inline client/app/components/calendar/event/Event.jsx:12]: You need to be consistent - either align always props to right or simple have one space

File: client/app/components/calendar/event/event_group.scss [hunk 4]
```diff
@@ -1,6 +1,5 @@
 
 .event-group {
-  align-items: stretch;
-  display: flex;
+  //display: flex;
```
Comment (adamzima) [inline client/app/components/calendar/event/event_group.scss]: ?

File: client/app/helpers/EventHelper.jsx [hunk 5]
```diff
@@ -1,15 +1,79 @@
...
-export function eventGroupContaining(events, timestamp) {
-  return events.find(group =>
-    group.some(eventStartsAt(timestamp))
-  );
+function eventsAssignedToColumns(eventsGroup) {
+  const columns = [];
+  eventsGroup.forEach(event => {
+    const columnForEvent = columns.find(column => (
+      column[column.length - 1].end_timestamp <= event.start_timestamp
+    ));
```
Comment (adamzima) [inline client/app/helpers/EventHelper.jsx]: No need to have parenthesis here

File: client/test/components/calendar/CalendarRow.spec.jsx [hunk 6]
```diff
@@ -3,19 +3,15 @@ import { shallow } from 'enzyme';
...
 describe('<CalendarRow />', () => {
-  const group1 = [Event.build(), Event.build()];
-  const group2 = [Event.build(), Event.build()];
-  const events = [group1, group2];
   const days = [new Date(), new Date(), new Date()];
   const unitEventLengthInSeconds = 1800;
   const onDelete = sinon.spy();
   let time = new Date();
-  const props = { events, days, unitEventLengthInSeconds, onDelete };
+  const props = { days, unitEventLengthInSeconds, onDelete };
```
Comment (adamzima) [inline client/test/components/calendar/CalendarRow.spec.jsx:14]: :+1:

File: client/app/helpers/EventHelper.jsx [hunk 7]
```diff
@@ -1,15 +1,79 @@
...
+  return columns;
 }
 
-export function eventsStartingAt(timestamp, group) {
-  return group.filter(eventStartsAt(timestamp));
+class Block {
+  constructor(event = null) {
+    this.blockEvents = [];
+    this.endTime = null;
+    if (event) {
```
Comment (mhib) [inline client/app/helpers/EventHelper.jsx]: event !== null?

File: client/app/components/calendar/event/event_group.scss (hunk 4 above)
Comment (adamzima) [inline client/app/components/calendar/event/event_group.scss]: ?

**Pull request merged**